pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -213,6 +213,10 @@ public void addField(FieldSpec fieldSpec) {
         throw new UnsupportedOperationException("Unsupported field type: " + fieldType);
     }
 
+    if (fieldSpec.getDataType() == null) {
+      throw new UnsupportedOperationException(String.format("'%s' field is missing 'dataType' property", columnName));
```

Comment:
Suggest modeling it as `IllegalArgumentException`. We can also use the `Preconditions` to simplify the check
```suggestion
      Preconditions.checkArgument(fieldSpec.getDataType() != null, "'%s' field is missing 'dataType' property", columnName);
```

Reply:
Oh, that's a cool idea, done!